sdk/core/azure-core-test/src/main/java/com/azure/core/test/policy/RecordNetworkCallPolicy.java

```diff
@@ -145,7 +145,16 @@ private void captureRequestHeaders(HttpHeaders requestHeaders, Map<String, Strin
 
         String contentType = response.getHeaderValue(CONTENT_TYPE);
         if (contentType == null) {
-            return Mono.just(responseData);
+            return response.getBodyAsByteArray().switchIfEmpty(Mono.just(new byte[0])).map(bytes -> {
+                if (bytes.length == 0) {
+                    return responseData;
+                }
+
+                String content = new String(bytes, StandardCharsets.UTF_8);
+                responseData.put("Content-Length", Integer.toString(content.length()));
```

Comment:
Define a constant for `Content-Length` just like you have for `CONTENT_TYPE` and `BODY`.

Reply:
Fixed